AUTHORING_GUIDE.md

```diff
@@ -1,207 +1,114 @@
+
+Source code files should always begin with an Apache 2.0 license header. See
+the instructions in the repo license file on [how to apply the Apache license
+to your work](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/LICENSE#L178-L201).
+For example, see the license header for the [Datastore client quickstart
+sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/datastore/cloud-client/quickstart.py#L1-L15).
 
 ### Shebang
 
 If, and only if, your sample application is a command-line application, then
-include a shebang as the first line. Separate the shebang from the rest of
-the application with a blank line. The shebang should always be:
+include a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) as the
+first line. Separate the shebang line from the rest of the application with
+a blank line. The shebang line for a Python application should always be:
 
 ```python
 #!/usr/bin/env python
 ```
 
-Don't include shebangs in web applications or test files.
+Don't include shebang lines in web applications or test files.
 
-### License header
+### Coding Style
 
-All samples should start with the following (modulo shebang line):
+All Python samples should follow the best practices defined in the
+[PEP 8 style guide](https://www.python.org/dev/peps/pep-0008/) and the
+[Google Python Style Guide](http://google.github.io/styleguide/pyguide.html).
+The automated linting process for Python samples uses
+[flake8](http://flake8.pycqa.org/en/latest/) to verify conformance to common
+Python coding standards, so the use of flake8 is recommended.
 
-```
-# Copyright 2020 Google, LLC.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#    http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-```
-
-### Module-level docstring
-
-All samples should contain a module-level docstring. For command-line
-applications, this docstring will be used as the summary when `-h` is passed.
-The docstring should be succinct and avoid repeating information
-available in readmes or documentation.
+If you prefer to use [pylint](https://www.pylint.org/), note that Python
+samples for this repo are not required to conform to pylint’s default settings
+outside the scope of PEP 8, such as the “too many arguments” or “too many
+local variables” warnings.
 
-Here's a simple docstring for a command-line application with straightforward
-usage:
+The use of [Black](https://pypi.org/project/black/) to standardize code
+formatting and simplify diffs is recommended, but optional.
```

Comment:
Would it be helpful to put that they should adjust the defaults of black to use `black -l 79` so that it matches PEP8?

Reply:
I would prefer we try to match what is being used on google-cloud-python/ python-{product} as to not be divergent. A few flake8 rules are ignored (https://github.com/googleapis/python-storage/blob/master/.flake8) and black is used with defaults.

Reply:
+1 to consistency with the client repos